### DISCUSSION TITLE
Long reinstall time due to old registry protocol. Specify sparse by default in .cargo/config.toml

### DISCUSSION THREAD

**Discussion by EriKWDev** [helix-editor/helix#8792] [General] [answered]
![image](https://github.com/helix-editor/helix/assets/19771356/44da190b-949f-4ff8-91d4-fd6fd87280b9)

The `rust-toolchain.toml` file locks the toolchain at version `1.65.0` meaning that whenever I need to update helix from source, cargo will use the old registry protocol method instead of the newer sparse protocol which only fetches relevant packages instead of all of them.

My internet connection is pretty poor and while this might not seem like a big issue, it is still annoying every time.

For example, if we in the `./.cargo/config.toml` file would include
```toml
[registries.crates-io]
protocol = "sparse"
```

I believe that would solve the problem

I also tried doing `rustup default 1.65.0` and clear the registry cache, and the older version of cargo seems to pickup this hint as well.

**Comment by pascalkuthe** [comment]
You could just remove the rusttoolchain file locally. We will update our MSRV once firefox does.  I don't think it's necessary to do a workaround until then

**Comment by archseer** [comment, marked as answer]
You can also configure this globally in `~/.cargo/config.toml`.
